### PR TITLE
Allow embedding custom UI config in index.html

### PR DESCRIPTION
**Pull request by yurishkuro**
For #489

```
$ cat yy.yaml
x: abcd
z:
  - a
  - b

$ go run ./cmd/standalone/main.go --span-storage.type=memory --query.ui-config yy.yaml
```

replaces

```
<!-- JAEGER_CONFIG = DEFAULT_CONFIG; -->
```

in `index.html` with

```
<!-- JAEGER_CONFIG = {"x":"abcd","z":["a","b"]}; -->
```

TODO:
  * [x] tests
  * [x] UI version that includes the config var in `index.html`

### CODE REVIEW

File: cmd/query/app/fixture/index.html [hunk 1]
```diff
@@ -2,4 +2,5 @@
 <html lang="en">
     <meta charset="UTF-8">
     <title>Test Page</title>
+    <!-- JAEGER_CONFIG=DEFAULT_CONFIG; -->
```
Comment (black-adder) [inline cmd/query/app/fixture/index.html]: change this to: "JAEGER_CONFIG = DEFAULT_CONFIG;" to be a more accurate fixture?

File: cmd/query/app/static_handler.go [hunk 2]
```diff
@@ -27,22 +34,66 @@ const (
...
+		return nil, nil
+	}
+	ext := filepath.Ext(uiConfig)
+	bytes, err := ioutil.ReadFile(uiConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Cannot read UI config file %v", uiConfig)
+	}
+
+	var c map[string]interface{}
+	var unmarshall func([]byte, interface{}) error
```
Comment (black-adder) [inline cmd/query/app/static_handler.go]: s/unmarshall/unmarshal

File: cmd/query/app/fixture/index.html (hunk 1 above)
Comment (yurishkuro) [reply cmd/query/app/fixture/index.html]: how is it more accurate?  The regexp ignores the whitespace.

File: cmd/query/app/static_handler.go [hunk 3]
```diff
@@ -27,22 +33,67 @@ const (
...
 	if staticAssetsRoot == "" {
 		staticAssetsRoot = defaultStaticAssetsRoot
 	}
 	if !strings.HasSuffix(staticAssetsRoot, "/") {
 		staticAssetsRoot = staticAssetsRoot + "/"
 	}
-	return &StaticAssetsHandler{staticAssetsRoot: staticAssetsRoot}
+	indexBytes, err := ioutil.ReadFile(staticAssetsRoot + "index.html")
+	if err != nil {
+		return nil, errors.Wrap(err, "Cannot read UI static assets")
```
Comment (pavolloffay) [inline cmd/query/app/static_handler.go]: Now it fails without UI. Before it was working. Workaround is to create empty `index.html` or do not include static handler if flag is empty 

https://github.com/jaegertracing/jaeger/pull/493 uses query as a docker container, when xdock is executed it builds all images, Now we have to build UI which takes significant time.

Comment (yurishkuro) [reply cmd/query/app/static_handler.go]: let's discuss on #497

**Pull request merged**